linxiangxi/server/src/server.cpp

```diff
@@ -0,0 +1,74 @@
+#include <ros/ros.h>
+#include <time.h>
+#include <client/client_message.h>
+#include <client/client_login.h>
+#include <vector>
+
+#define SUBSCRIBER_SIZE 10
+
+#define LOGIN 0
+#define LOGOUT 1
+#define NUL 2
+#define SUCCESS 3
+#define FAILED 4
+
+typedef struct{
+    ros::Subscriber client_sub;
+    std::string client_name;
+}Client_info;
+
+ros::NodeHandle * server_node_handle_ptr = nullptr;
+std::vector<Client_info> client_storage;
+
+void client_message_callback(const client::client_message & client_message){
+    char output_tmp[100];
+    time_t _time = client_message.current_time.sec;
+    strftime(output_tmp, sizeof(output_tmp), 
```

Comment:
**[cpplint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]